src/Interpreters/DictionaryJoinAdapter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <Core/Block.h>
+#include <Columns/ColumnVector.h>
+#include <Dictionaries/IDictionary.h>
+#include <Storages/IKVStorage.h>
+
+namespace DB
+{
+
+/// Used in join with dictionary to provide sufficient interface to DirectJoin
+class DictionaryJoinAdapter : public IKeyValueStorage
```

Comment:
Consider to extend IDictionary to support IKeyValueStorage interface without creating adaptor